akka-serialization-jackson/src/test/scala/akka/serialization/jackson/ScalaTestEventMigration.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.serialization.jackson
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.node.IntNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+
+class ScalaTestEventMigrationV2 extends JacksonMigration {
+  override def currentVersion = 2
+
+  override def transformClassName(fromVersion: Int, className: String): String =
+    classOf[ScalaTestMessages.Event2].getName
+
+  override def transform(fromVersion: Int, json: JsonNode): JsonNode = {
+    val root = json.asInstanceOf[ObjectNode]
+    root.set[JsonNode]("field1V2", root.get("field1"))
+    root.remove("field1")
+    root.set[JsonNode]("field2", IntNode.valueOf(17))
+    root
+  }
+
+}
+
+class ScalaTestEventMigrationV2WithV3 extends JacksonMigration {
+  override def currentVersion = 2
+
+  override def supportedForwardVersion: Int = 3
+
+  // Always produce the type of the currentVersion. When fromVersion is lower,
+  // transform will lift it. When fromVersion is higher, transform will downcast it.
+  override def transformClassName(fromVersion: Int, className: String): String =
+    classOf[ScalaTestMessages.Event2].getName
+
+  override def transform(fromVersion: Int, json: JsonNode): JsonNode = {
+    if (fromVersion < 2) {
+      val root = json.asInstanceOf[ObjectNode]
+      root.set[JsonNode]("field1V2", root.get("field1"))
+      root.remove("field1")
+      root.set[JsonNode]("field2", IntNode.valueOf(17))
+      return root
```

Comment:
Surprising to see `return`. Separate `private` methods make intention more obvious.

Reply:
Good catch. Not only it's not idiomatic, but it'll be wrong when a new migration is added (this comment made me realize there's a bug in `ScalaTestEventMigrationV3` below).